core/fuzzer.h

```diff
@@ -46,15 +76,62 @@ class P4RuntimeFuzzer {
     virtual uint32_t producePriority(
         const google::protobuf::RepeatedPtrField<p4::config::v1::MatchField> &matchFields);
 
-    virtual std::optional<p4::v1::FieldMatch> produceMatchField(p4::config::v1::MatchField &match,
-                                                                int fieldId);
+    virtual p4::v1::FieldMatch produceMatchField(p4::config::v1::MatchField &match, int fieldId);
     virtual p4::v1::TableEntry produceTableEntry(
         const p4::config::v1::Table &table,
         const google::protobuf::RepeatedPtrField<p4::config::v1::Action> &actions);
 
-    virtual InitialP4RuntimeConfig produceInitialConfig() = 0;
+    virtual InitialConfig produceInitialConfig() = 0;
+
+    virtual UpdateSeries produceUpdateTimeSeries() = 0;
+};
+
+class BFRuntimeInitialConfig : public InitialConfig {
+ public:
+    BFRuntimeInitialConfig() = default;
+    std::vector<bfrt_proto::WriteRequest> requests;
+};
+
+class BFRuntimeUpdateSeries : public UpdateSeries {
+    std::vector<std::pair<uint64_t, bfrt_proto::WriteRequest>> updates;
+};
+
+class BFRuntimeFuzzer : public RuntimeFuzzer {
+ public:
+    explicit BFRuntimeFuzzer(const ProgramInfo &programInfo)
+        : RuntimeFuzzer(FuzzerType::BFRuntime, programInfo) {}
+
+    BFRuntimeFuzzer(const BFRuntimeFuzzer &) = default;
+    BFRuntimeFuzzer(BFRuntimeFuzzer &&) = delete;
+    BFRuntimeFuzzer &operator=(const BFRuntimeFuzzer &) = default;
+    BFRuntimeFuzzer &operator=(BFRuntimeFuzzer &&) = delete;
+    virtual ~BFRuntimeFuzzer() = default;
+
+    virtual bfrt_proto::KeyField_Exact produceKeyField_Exact(int bitwidth);
```

Comment:
Would be nice to have triple-slash comments for these functions. Maybe the original does not have them either but it is a nice invariant to keep.

Reply:
I saw the triple-slash comments in .cpp files. Is it better to put in .h or .cpp?

Reply:
header